File: backend/items/entities/promo.go
package entities

type PromotStruct struct {
	Id            string         `json:"id" bson:"_id,omitempty"`
	Name          string         `json:"name" bson:"name"`
	Description   string         `json:"description" bson:"description"`
	DiscountType  string         `json:"discount_type" bson:"discount_type"` // percentage, fixed, bogo
	DiscountValue float64        `json:"discount_value" bson:"discount_value"`
	Code          string         `json:"code" bson:"code"`
	StartDate     string         `json:"start_date" bson:"start_date"`
	EndDate       string         `json:"end_date" bson:"end_date"`
	Products      []ProductOrder `json:"products" bson:"products"`
	MinPurchase   float64        `json:"min_purchase" bson:"min_purchase"`
	Status        string         `json:"status" bson:"status"` // active, inactive, expired
	CreatedAt     string         `json:"created_at" bson:"created_at"`
	UpdatedAt     string         `json:"updated_at" bson:"updated_at"`
}

type PromoResponse struct {
	Id            string         `json:"id" bson:"_id,omitempty"`
	Name          string         `json:"name" bson:"name"`
	Description   string         `json:"description" bson:"description"`
	DiscountType  string         `json:"discount_type" bson:"discount_type"` // percentage, fixed, bogo
	DiscountValue float64        `json:"discount_value" bson:"discount_value"`
	Code          string         `json:"code" bson:"code"`
	StartDate     string         `json:"start_date" bson:"start_date"`
	EndDate       string         `json:"end_date" bson:"end_date"`
	Products      []ProductOrder `json:"products" bson:"products"`
	MinPurchase   float64        `json:"min_purchase" bson:"min_purchase"`
}

File: mobile/lib/type/promo.dart
import 'package:trinity/type/product.dart';

class Promo {
  final String id;
  final String name;
  final String description;
  final String discountType;
  final double discountValue;
  final String code;
  final String startDate;
  final String endDate;
  final List<Product> products;
  final double minPurchase;

  Promo({
    required this.id,
    required this.name,
    required this.description,
    required this.discountType,
    required this.discountValue,
    required this.code,
    required this.startDate,
    required this.endDate,
    required this.products,
    required this.minPurchase,
  });

  factory Promo.fromJson(Map<String, dynamic> json) {
    return Promo(
      id: json['id'] ?? '',
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      discountType: json['discount_type'] ?? '',
      discountValue: (json['discount_value'] ?? 0).toDouble(),
      code: json['code'] ?? '',
      startDate: json['start_date'] ?? '',
      endDate: json['end_date'] ?? '',
      products: (json['products'] as List<dynamic>?)
          ?.map((p) => Product.fromJson(p as Map<String, dynamic>))
          .toList() ?? [],
      minPurchase: (json['min_purchase'] ?? 0).toDouble(),
    );
  }

  @override
  String toString() {
    return 'Promo(id: $id, name: $name, description: $description, discountType: $discountType, discountValue: $discountValue, code: $code, startDate: $startDate, endDate: $endDate, minPurchase: $minPurchase, products: $products)';
  }
}